### PR TITLE
Add CODECOV_TOKEN to ci.yml to fix Codecov uploads

### LINKED ISSUE #703: Codecov upload failures
Codecov uploads have been silently failing for around 25 days, see [the last reported commits](https://app.codecov.io/gh/UXARRAY/uxarray/commits) and [your logs here](https://github.com/UXARRAY/uxarray/actions/runs/8057964419/job/22010007742#step:10:21)

We fixed the same issue in geocat-comp (https://github.com/NCAR/geocat-comp/pull/566) by following this documentation: https://docs.codecov.com/docs/adding-the-codecov-token

### PR DESCRIPTION
**Pull request by erogluorhan**
Closes #703

## Overview
This PR adds "CODECOV_TOKEN" into ci.yml (CODECOV_TOKEN has just been added to the repo secrets)

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,7 +71,10 @@ jobs:
           python -m pytest test -v --cov=./uxarray --cov-report=xml
 
       - name: Upload code coverage to Codecov
+        if: github.repository == 'NCAR/geocat-comp'
```
Comment (philipc2) [inline .github/workflows/ci.yml:74]: I believe this should be the following? 

```suggestion
        if: github.repository == 'UXARRAY/uxarray'
```

**Review by philipc2 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (erogluorhan) [reply .github/workflows/ci.yml:74]: Oh, a good proof of one should avoid copy-paste late in the evening :) Thanks for catching!

**Review by philipc2 — APPROVED** [review]
Thanks for putting this together!

**Comment by philipc2** [comment]
Looks to be working as expected now!

https://app.codecov.io/gh/UXARRAY/uxarray/commits

**Pull request merged**